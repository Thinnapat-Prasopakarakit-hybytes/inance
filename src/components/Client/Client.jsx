import React from "react";
import "./Client.scss";
//import ReactDOMServer from "react-dom/server";
import client1 from "../../assets/images/client-1.jpg";
import client2 from "../../assets/images/client-2.jpg";
import OwlCarousel from "react-owl-carousel";
import "owl.carousel/dist/assets/owl.carousel.css";
import "owl.carousel/dist/assets/owl.theme.default.css";
import { FaQuoteLeft, FaStar } from "react-icons/fa";

const Client = () => {
  const clients = [
    {
      image: client1,
      info: {
        name: "Jorch morik",
        rating: 5,
      },
      text: "chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum",
    },
    {
      image: client2,
      info: {
        name: "Jorch morik",
        rating: 5,
      },
      text: "chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum",
    },
    {
      image: client1,
      info: {
        name: "Jorch morik",
        rating: 5,
      },
      text: "chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum",
    },
    {
      image: client2,
      info: {
        name: "Jorch morik",
        rating: 5,
      },
      text: "chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum",
    },
  ];

  const options = {
    loop: true,
    margin: 10,
    nav: true,
    navText: [
      //React Icons
      // ReactDOMServer.renderToString(<FaLongArrowAltLeft />),
      // ReactDOMServer.renderToString(<FaLongArrowAltRight />),

      //Font Awesome
      '<i class="fa fa-long-arrow-left"></i>',
      '<i class="fa fa-long-arrow-right"></i>',
    ],
    dots: false,
    autoplay: true,
    autoplayTimeout: 3000,
    autoplayHoverPause: true,
    rewind: true,
    responsive: {
      0: {
        items: 1,
      },
      768: {
        items: 2,
      },
    },
    navElement: "div",
    smartSpeed: 250,
    items: 1,
  };

  return (
    <section className="client_section">
      <div className="container">
        <div className="heading_container heading_center">
          <h2>What Our Clients Say</h2>
        </div>
        <div className="carousel-wrap layout_padding2-top">
          <OwlCarousel {...options}>
            {clients.map((client, index) => (
              <div key={index} className="item">
                <div className="box">
                  <div className="client_id">
                    <div className="img-box">
                      <img src={client.image} alt={client.info.name} />
                    </div>
                    <div className="client_detail">
                      <div className="client_info">
                        <h6>{client.info.name}</h6>
                        <div className="rating">
                          {[...Array(client.info.rating)].map((_, i) => (
                            <FaStar key={i} className="rating_star" />
                          ))}
                        </div>
                      </div>
                      <FaQuoteLeft />
                    </div>
                  </div>
                  <div className="client_text">
                    <p>{client.text}</p>
                  </div>
                </div>
              </div>
            ))}
          </OwlCarousel>
        </div>
      </div>
    </section>
  );
};

export default Client;
