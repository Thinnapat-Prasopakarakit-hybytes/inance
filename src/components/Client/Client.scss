@use "../../styles/partials/variables" as *;
@use "../../styles/partials/mixins" as *;

.client_section {
  .heading_container {
    align-items: center;
    text-align: center;
  }

  .box {
    margin: 15px;
    box-shadow: 0 0 10px 0 rgba($color: #000000, $alpha: 0.2);
    padding: 25px;

    .client_id {
      display: flex;

      .img-box {
        width: 125px;
        min-width: 125px;
        margin-right: 15px;

        img {
          width: 100%;
          border-radius: 100%;
        }

        [dir="rtl"] & {
          margin-right: 0;
          margin-left: 15px;
        }
      }

      .client_detail {
        display: flex;
        align-items: center;
        justify-content: space-between;
        flex: 1;
        padding-right: 25px;

        [dir="rtl"] & {
          padding-right: 0;
          padding-left: 25px;
        }

        .client_info {
          h6 {
            margin-bottom: 0;
          }

          i {
            color: #fbaf5d;
          }
        }
        .rtl-quote {
          transform: scaleX(-1);
        }
      }
    }

    .client_text {
      margin-top: 20px;
    }
  }

  .rating_star {
    color: #fbaf5d;
  }

  .owl-carousel {
    .owl-nav {
      display: flex;
      justify-content: center;
      margin-top: 20px;

      [dir="rtl"] & {
        flex-direction: row-reverse;
      }

      .owl-prev,
      .owl-next {
        width: 50px;
        height: 50px;
        background-color: $primary1;
        outline: none;
        margin: 0 20px;
        color: $white;
        display: flex;
        align-items: center;
        justify-content: center;
      }

      [dir="rtl"] & {
        transform: scaleX(-1);
      }
    }
  }
}

[dir="rtl"] {
  .owl-carousel {
    direction: ltr;

    .owl-item {
      direction: rtl;
    }
  }
}
