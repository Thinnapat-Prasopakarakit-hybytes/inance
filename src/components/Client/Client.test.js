import React from "react";
import { render, screen } from "@testing-library/react";
import Client from "./Client";
import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";

jest.mock("react-icons/fa", () => ({
  FaQuoteLeft: () => <div data-testid="quote-icon" />,
  FaStar: () => <div data-testid="star-icon" />,
}));

jest.mock("react-owl-carousel", () => {
  return function DummyOwlCarousel({ children }) {
    return <div data-testid="owl-carousel">{children}</div>;
  };
});

const renderClient = () => {
  return render(
    <BrowserRouter>
      <Client />
    </BrowserRouter>
  );
};

describe("Client Component", () => {
  it("should render the heading", () => {
    renderClient();
    expect(screen.getByText("What Our Clients Say")).toBeInTheDocument();
  });

  it("should render client testimonials", () => {
    renderClient();
    const clientNames = screen.getAllByText("Jorch morik");
    expect(clientNames).toHaveLength(4);
  });

  it("should render rating stars for each client", () => {
    renderClient();
    const stars = screen.getAllByTestId("star-icon");
    expect(stars).toHaveLength(20);
  });

  it("should render quote icons", () => {
    renderClient();
    const quoteIcons = screen.getAllByTestId("quote-icon");
    expect(quoteIcons).toHaveLength(4);
  });

  it("should render client testimonial text", () => {
    renderClient();
    const testimonialText = screen.getAllByText(/chunks as necessary/i);
    expect(testimonialText).toHaveLength(4);
  });

  it("should match snapshot", () => {
    const { asFragment } = renderClient();
    expect(asFragment()).toMatchSnapshot();
  });
});
