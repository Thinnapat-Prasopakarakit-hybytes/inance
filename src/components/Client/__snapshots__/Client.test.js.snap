// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client Component should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="client_section"
  >
    <div
      class="container"
    >
      <div
        class="heading_container heading_center"
      >
        <h2>
          What Our Clients Say
        </h2>
      </div>
      <div
        class="carousel-wrap layout_padding2-top"
      >
        <div
          data-testid="owl-carousel"
        >
          <div
            class="item"
          >
            <div
              class="box"
            >
              <div
                class="client_id"
              >
                <div
                  class="img-box"
                >
                  <img
                    alt="Jorch morik"
                    src="mock-image-path"
                  />
                </div>
                <div
                  class="client_detail"
                >
                  <div
                    class="client_info"
                  >
                    <h6>
                      Jorch morik
                    </h6>
                    <div
                      class="rating"
                    >
                      <div
                        data-testid="star-icon"
                      />
                      <div
                        data-testid="star-icon"
                      />
                      <div
                        data-testid="star-icon"
                      />
                      <div
                        data-testid="star-icon"
                      />
                      <div
                        data-testid="star-icon"
                      />
                    </div>
                  </div>
                  <div
                    data-testid="quote-icon"
                  />
                </div>
              </div>
              <div
                class="client_text"
              >
                <p>
                  chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum
                </p>
              </div>
            </div>
          </div>
          <div
            class="item"
          >
            <div
              class="box"
            >
              <div
                class="client_id"
              >
                <div
                  class="img-box"
                >
                  <img
                    alt="Jorch morik"
                    src="mock-image-path"
                  />
                </div>
                <div
                  class="client_detail"
                >
                  <div
                    class="client_info"
                  >
                    <h6>
                      Jorch morik
                    </h6>
                    <div
                      class="rating"
                    >
                      <div
                        data-testid="star-icon"
                      />
                      <div
                        data-testid="star-icon"
                      />
                      <div
                        data-testid="star-icon"
                      />
                      <div
                        data-testid="star-icon"
                      />
                      <div
                        data-testid="star-icon"
                      />
                    </div>
                  </div>
                  <div
                    data-testid="quote-icon"
                  />
                </div>
              </div>
              <div
                class="client_text"
              >
                <p>
                  chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum
                </p>
              </div>
            </div>
          </div>
          <div
            class="item"
          >
            <div
              class="box"
            >
              <div
                class="client_id"
              >
                <div
                  class="img-box"
                >
                  <img
                    alt="Jorch morik"
                    src="mock-image-path"
                  />
                </div>
                <div
                  class="client_detail"
                >
                  <div
                    class="client_info"
                  >
                    <h6>
                      Jorch morik
                    </h6>
                    <div
                      class="rating"
                    >
                      <div
                        data-testid="star-icon"
                      />
                      <div
                        data-testid="star-icon"
                      />
                      <div
                        data-testid="star-icon"
                      />
                      <div
                        data-testid="star-icon"
                      />
                      <div
                        data-testid="star-icon"
                      />
                    </div>
                  </div>
                  <div
                    data-testid="quote-icon"
                  />
                </div>
              </div>
              <div
                class="client_text"
              >
                <p>
                  chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum
                </p>
              </div>
            </div>
          </div>
          <div
            class="item"
          >
            <div
              class="box"
            >
              <div
                class="client_id"
              >
                <div
                  class="img-box"
                >
                  <img
                    alt="Jorch morik"
                    src="mock-image-path"
                  />
                </div>
                <div
                  class="client_detail"
                >
                  <div
                    class="client_info"
                  >
                    <h6>
                      Jorch morik
                    </h6>
                    <div
                      class="rating"
                    >
                      <div
                        data-testid="star-icon"
                      />
                      <div
                        data-testid="star-icon"
                      />
                      <div
                        data-testid="star-icon"
                      />
                      <div
                        data-testid="star-icon"
                      />
                      <div
                        data-testid="star-icon"
                      />
                    </div>
                  </div>
                  <div
                    data-testid="quote-icon"
                  />
                </div>
              </div>
              <div
                class="client_text"
              >
                <p>
                  chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
